### PR TITLE
Update code with API

### PR DESCRIPTION
**Pull request by chirag-456**

### CODE REVIEW

**Review by bballboy8 — DISMISSED** [review]

File: controller/authController.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ exports.register = (req, res) => {
     // Create new user
     createUser(email, password, (err, result) => {
       if (err) {
-        return res.status(500).json({ message: 'Server Error' });
+        return res.status(500).json({ message: 'Server Error need to check ' });
```
Comment (bballboy8) [inline controller/authController.js]: Correct the english here

File: controller/creditCardController.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // controller/creditCardController.js
+const stripe = require('stripe')(process.env.STRIPE_SANDBOX);
```
Comment (bballboy8) [inline controller/creditCardController.js:6]: Check for the existence of this key before setting it

File: controller/creditCardController.js [hunk 3]
```diff
@@ -56,3 +57,35 @@ exports.getCreditCardById = (req, res) => {
     res.status(200).json({ card });
   });
 };
+
+
+exports.chargePayment = async (req, res) => {
+  const { token, amount } = req.body;
```
Comment (bballboy8) [inline controller/creditCardController.js]: Need validation for token and amount

File: routes/creditCardRoutes.js [hunk 4]
```diff
@@ -14,4 +14,6 @@ router.get('/all', getAllCreditCards);
 // Get a specific credit card by ID (requires token)
 router.get('/:id', getCreditCardById);
 
+router.post('/charge-payment', chargePayment);
```
Comment (bballboy8) [inline routes/creditCardRoutes.js]: Is there authentication middleware needed here?

File: package.json [hunk 5]
```diff
@@ -24,7 +24,8 @@
     "express": "^4.21.2",
     "jsonwebtoken": "^9.0.2",
     "mongoose": "^8.9.0",
-    "mysql2": "^3.11.5"
```
Comment (bballboy8) [inline package.json:27]: Everything should enforce a specific version of node.js dependency

**Review by bballboy8 — COMMENTED** [review]
Also extract all business logic into server modules so this is easier to read and modify

File: models/creditCardModel.js [hunk 6]
```diff
@@ -1,51 +1,100 @@
...
-        // Card is already registered, return an error message
-        return callback(null, { message: 'Card already registered' });
+  // Validate and sanitize card number and CVV code (ensure they are numeric and valid)
+  cardNo = cardNo.replace(/\D/g, ''); // Remove any non-digit characters
+  cvvCode = cvvCode.replace(/\D/g, ''); // Remove any non-digit characters
+
+  // Step 2: Input validation
+
+  // Validate the card number using Luhn algorithm (for example, using validator library or custom logic)
+  if (!validator.isLength(cardNo, { min: 13, max: 19 })) {
```
Comment (bballboy8) [inline models/creditCardModel.js]: Use Luhn's algorithm: https://www.npmjs.com/package/luhn

**Review by bballboy8 — CHANGES_REQUESTED** [review]

File: middleware/authenticate.js [hunk 7]
```diff
@@ -1,20 +1,31 @@
...
-    return res.status(401).json({ message: 'No token provided' });
-  }
-  jwt.verify(token, process.env.JWT_SECRET, (err, decoded) => {
-    if (err) {
-      return res.status(401).json({ message: 'Invalid or expired token' });
+  try {
+    // Check if the 'Authorization' header is present and starts with 'Bearer' (case-insensitive)
+    const authHeader = req.headers['authorization'];
+    if (!authHeader || !authHeader.toLowerCase().startsWith('bearer ')) {
+      return res.status(401).json({ message: 'Authorization header must be in Bearer token format' });
```
Comment (bballboy8) [inline middleware/authenticate.js]: "Bearer token format" shouldn't be mentioned in a log. It discloses too much information about the security of the application.

File: models/creditCardModel.js [hunk 8]
```diff
@@ -1,51 +1,124 @@
...
+    return callback({ message: 'Provide correct card no.' }, null);
+  }
+
+
+  if (!validator.isNumeric(cardNo)) {
+    return callback({ message: 'Card number must only contain numeric characters.' }, null);
+  }
+
+  // Validate expiry date format (MM/YY)
+  if (!validator.matches(expiryDate, /^(0[1-9]|1[0-2])\/\d{2}$/)) {
```
Comment (bballboy8) [inline models/creditCardModel.js]: Use Validator library instead of regex

Comment (chirag-456) [reply models/creditCardModel.js]: I have used the NPM library now to resolve this

File: middleware/authenticate.js (hunk 7 above)
Comment (chirag-456) [reply middleware/authenticate.js]: Issues is resolved and token stored in Encrypted format

**Review by bballboy8 — APPROVED** [review]

**Pull request merged**